Document transformation meta data (#108)
xataio/mdx-docs: 040-SDK/160-image-transformations.mdx

  options={{ rotate: 180, height: 100 }}
/>

## Record based transformations

> When querying a `file` of `file[]` column off a record, you can make a direct transformation on any files contained.

Reviewer: ```suggestion
When querying a `file` of `file[]` column from a record, you can make a direct transformation on any of the files it contains.
```

<TabbedCode tabs={['TypeScript', 'URL']}>

```ts
// Get a specific record from the "gallery" database
const record = await xata.db.gallery.read('record_id');

// Get a new url and signedUrl based on transformations applied to
// an image in the "imageColumnName"
//
// Also grab the related meta data (height, width...etc) of the new image

> const { url, signedUrl, metaDataUrl, signedMetaDataUrl } = record.imageColumnName.transform({

Reviewer: ```suggestion
const { url, signedUrl, metadataUrl, signedMetadataUrl } = record.imageColumnName.transform({
```

  height: 100,
  rotate: 180,
  format: 'webp'
});
```

```txt
https://us-east-1.storage.xata.sh/transform/height=50,format=webp/4u1fh2o6p10blbutjnphcste94
```

</TabbedCode>

## URL based transformations

It is also possible to make transformations directly on any given image URL without knowledge of the record it is attached to.

For TypeScript, this is done client-side with the `transformImage()` helper in `@xata.io/client`. Performing client-side transformations can be useful for generating images conditionally based on media queries within a browser. With this method, only a `url` for an existing Xata image is needed.

> <TabbedCode tabs={['TypeScript','Python', 'URL']}>

Reviewer: ```suggestion
<TabbedCode tabs={['TypeScript', 'Python', 'URL']}>
```

</TabbedCode>

## Post-transformation metadata

> Once a transformation occurs, it is common to need the dimensions of the new image. `metaDataUrl` and `signedMetaUrl` (if using signed URLs) are available when making transformations from a file on a record.

Reviewer: ```suggestion
Once a transformation occurs, it is common to need the dimensions of the new image. `metadataUrl` and `signedMetadataUrl` (if using signed URLs) are available when making transformations from a file on a record.
```

```ts
// Get a specific record from the "gallery" database
const record = await xata.db.gallery.read('record_id');

// Also grab the related meta data (height, width...etc) of the new image

> const { url, metaDataUrl } = record.imageColumnName.transform({

Reviewer: ```suggestion
const { url, metadataUrl } = record.imageColumnName.transform({
```

  height: 100,
  rotate: 180,
  format: 'webp'
});

> console.log('height of new image', metaDataUrl.height); // 100px

Reviewer: Nope, you still need to fetch the result. It's a URL not the response of the URL
